lib/index.js

```diff
@@ -51,6 +51,16 @@ exports.getStoreInstances = function(db, fn) {
     .find({})
     .toArray(function(err, docs) {
       if (err) {
+        if (isNotAuthorized(err)) {
+          debug('not authorized to get metadata to '
+            + 'discover replicaset instances. omitting.');
+
+          var serverUrl = db.serverConfig.s.options.url;
```

Comment:
@Machyne better to make this `_.has(db, 'serverConfig.s.options.url')` just because it's a driver internal and would suck if that gets changed on us :)